pkg/cluster/mpi/istiopeerauthentication.go

```diff
@@ -38,3 +39,32 @@ func (s *istioPeerAuthenticationDS) Enabled() bool {
 func (s *istioPeerAuthenticationDS) Delete() bool {
 	return s.mpi.Spec.MutualTLSMode == ""
 }
+
+func IstioClientPeerAuthentication(enabled bool) core.Component {
+	return components.IstioPeerAuthentication(func(obj client.Object) components.IstioPeerAuthenticationDataSource {
+		return &istioClientPeerAuthenticationDS{mpi: objToMPICluster(obj), enabled: enabled}
+	})
+}
+
+type istioClientPeerAuthenticationDS struct {
+	mpi     *dcv1alpha1.MPICluster
+	enabled bool
+}
+
+func (s *istioClientPeerAuthenticationDS) PeerAuthInfo() *istio.PeerAuthInfo {
+	return &istio.PeerAuthInfo{
+		Name:      meta.InstanceName(s.mpi, ComponentClient),
+		Namespace: s.mpi.Namespace,
+		Labels:    meta.StandardLabels(s.mpi),
+		Selector:  s.mpi.Spec.NetworkPolicy.ClientLabels,
+		Mode:      authenticationv1alpha1.MutualTls_PERMISSIVE.String(),
+	}
+}
+
+func (s *istioClientPeerAuthenticationDS) Enabled() bool {
```

Comment:
nit: possible to use the internal bool /  setting that results in this annotations value rather than the annotation itself? Would be more resilient to changes in API in the future

Reply:
That is not currently passed in the CR. Principally, we are more worried about whether the sidecar is being injected than how that is configured. Unfortunately the annotation is probably the most stable way to determine that.